app/models/schedule.rb

```diff
@@ -0,0 +1,15 @@
+class Schedule < ApplicationRecord
+  validates :start_time, presence: true
+  validate :start_time, :date_cannot_be_before_start_time
+  validates :end_time, presence: true, comparison: { greater_than: :start_time }
+  validates :day, presence: true
+
+  private
+
+  def date_cannot_be_before_start_time
+    if start_time >= end_time
+      errors.add(:date, "開始時刻、または終了時刻が間違っています")
```

Comment:
エラーメッセージは極力それを見たユーザーがどうしたらいいかわかる内容にしてあげるといいかも

```suggestion
      errors.add(:date, "開始時刻は終了時刻以前に設定してください")
```